docs/understanding-tracking-design/json-schema-type-casting-rules/index.md

```diff
@@ -0,0 +1,482 @@
+---
+title: "How schemas translate to database types"
+date: "2021-12-30"
+sidebar_position: 100
+hide_table_of_contents: true
+---
+
+```mdx-code-block
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+```
+
+_The row order in this table is important.  Type lookup stops after first match is found scanning from top to bottom (with the single exception of "null" - first row in the table)_
+<Tabs groupId="type-casting">
+  <TabItem value="redshift" label="Redshift" default>
+
+<table>
+<thead>
+<td>Json Schema</td>
+<td>Redshift Type</td>
+<td>Notes</td>
+</thead>
+<tbody>
+<tr>
+<td >
+
+  ```json
+{
+    "type": ["null", T1, ...]
+}
+```
+
+OR
+
+  ```json
+{
+    "enum": ["null", E1, ...]
+}
+```
+
+</td>
+<td >
+<code>NULLABLE</code>
+</td>
+<td><code>"null"</code> is not considered for type casting logic. Only for nullability constraint. Type lookup will continue down the table.
+</td>
+</tr>
+<tr>
+<td>
+
+  ```json
+{
+    "enum": [E1, E2, ...]
+}
+```
+
+</td>
+<td><code>VARCHAR(M)</code></td>
+<td> Where <code>M</code> is maximum size of <code>json.stringify(E*)</code><br/><br/>
+The <code>enum</code> can contain more than <b>one</b> JavaScript type: <code>string</code>, <code>number|integer</code>, <code>boolean</code>.<br/>
```

Comment:
Will strings be quoted, as per JSON encoding? That would be important to call out in the notes.